### PR TITLE
Implement TTS WS

### PR DESCRIPTION
**Pull request by dvonthenen**
Implements/Addresses: https://github.com/deepgram/deepgram-dotnet-sdk/issues/306

This PR implements TTS WS. In this PR, I restructured the Speak/TTS clients because we are introducing a WS Client, and the REST client was at the root of the `Deepgram.<Clients | Models>.Speak.v1` folder would place the REST and WS client in the same namespace. This is awkward and can be confusing. This follows the Go SDK, which places each into the respective namespaces:
- `Deepgram.<Clients | Models>.Speak.v1.REST`
- `Deepgram.<Clients | Models>.Speak.v1.Websocket`

The REST client has been renamed `SpeakClient` -> `SpeakRESTClient`. References to the old (now obsolete) REST client in `Deepgram.<Clients | Models>.Speak.v1` references have been marked obsolete but still exist for backward compatibility. The references to the examples for the Speak REST clients are intentionally pointing to the obsolete namespaces to prove backwards compatibility. These will be updated in a subsequent PR.

This implements a simple example to demonstrate the functionality and updates the Speak REST unit tests.

The README was **NOT** updated because they need to be updated for `Listen` and `Speak`. Will do so in a subsequent PR.

## Summary by CodeRabbit

- **New Features**
  - Introduced REST and WebSocket clients for interacting with the Deepgram API.
  - Added support for streaming and asynchronous audio transcription with callback functionalities.
  - Implemented various WebSocket events and message handling.

- **Enhancements**
  - Improved logging messages for API base addresses.
  - Added new constants for better consistency and management across the SDK.

- **Deprecations**
  - Deprecated `ISpeakClient` in favor of `ISpeakRESTClient`.
  - Deprecated `AsyncResponse` class, now derived from `REST.AsyncResponse`.

- **Documentation**
  - Updated XML documentation comments for new classes and methods.

### CODE REVIEW

**Comment by dvonthenen** [comment]
@coderabbitai review

File: examples/speak/websocket/simple/Speak.csproj [hunk 1]
```diff
@@ -15,5 +15,8 @@
 	  <ProjectReference Include="..\..\..\..\Deepgram\Deepgram.csproj" />
 	</ItemGroup>
 
+    <ItemGroup>
+        <Using Include="Deepgram" />
+    </ItemGroup>
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply examples/speak/websocket/simple/Speak.csproj:20]: Ignore this suggestion.

File: Deepgram/Models/Speak/v1/SyncResponse.cs [hunk 2]
```diff
@@ -4,72 +4,17 @@
...
+// This class provides the SyncResponse implementation for the Deepgram API
+//
+// Deprecated: This class is deprecated. Use the namespace `Deepgram.Models.Speak.v1.REST` package instead.
+// This will be removed in a future release.
+//
+// This class is frozen and no new functionality will be added.
+// *********** WARNING ***********
+/// </summary>
+[Obsolete("Please use Deepgram.Models.Speak.v1.REST.SyncResponse instead", false)]
+public record SyncResponse : REST.SyncResponse
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Speak/v1/SyncResponse.cs:18]: This is intentional. This is to allow users to still user those classes without needing to transition right away. Ignoring.

File: Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs [hunk 3]
```diff
@@ -0,0 +1,875 @@
...
+    /// </summary>
+    /// <returns>True if successful</returns>
+    public bool Subscribe(EventHandler<ErrorResponse> eventHandler)
+    {
+        lock (_mutexSubscribe)
+        {
+            _errorReceived += (sender, e) => eventHandler(sender, e);
+        }
+        return true;
+    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs:271]: This actually ended up producing a compiler error. Still investigating how to do this without messing up the event notification mechanism.

File: Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs [hunk 4]
```diff
@@ -0,0 +1,875 @@
...
+
+    ///// <summary>
+    /////  This method tells Deepgram to initiate the close server-side.
+    ///// </summary>
+    public void Close()
+    {
+        CommandSource commandSource = new CommandSource(Constants.Close);
+        byte[] byteArray = Encoding.UTF8.GetBytes(commandSource.ToString());
+        Send(byteArray);
+    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs:326]: SendMessage just places it in the queue and doesnt actually send. Ignore this.

File: Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs [hunk 5]
```diff
@@ -0,0 +1,875 @@
...
+            Log.Verbose("SpeakClient.Stop", "LEAVE");
+        }
+        catch (Exception ex)
+        {
+            Log.Error("Stop", $"{ex.GetType()} thrown {ex.Message}");
+            Log.Verbose("Stop", $"Excepton: {ex}");
+            Log.Verbose("SpeakClient.Stop", "LEAVE");
+            throw;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs:782]: This does clean up properly.

File: Deepgram/Clients/Interfaces/v1/ISpeakRESTClient.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+{
+    #region NoneCallBacks
+    /// <summary>
+    /// Speaks a file using the provided stream
+    /// </summary>
+    /// <param name="source">file is the form of a stream <see cref="Stream"/></param>
+    /// <param name="speakSchema">Options for the transcription <see cref="SpeakSchema"/></param>
+    /// <returns><see cref="SyncResponse"/></returns>
+    public Task<SyncResponse> ToStream(TextSource source, SpeakSchema? speakSchema, CancellationTokenSource? cancellationToken = default,
+        Dictionary<string, string>? addons = null, Dictionary<string, string>? headers = null);
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Interfaces/v1/ISpeakRESTClient.cs:24]: Yup, understood but we need a mechanism to cancel independently based on a timer.

File: Deepgram/Models/Speak/v1/WebSocket/v1/CommandSource.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    [JsonPropertyName("type")]
+    public string? Type { get; set; } = text;
+
+    /// <summary>
+    /// Override ToString method to serialize the object
+    /// </summary>
+    public override string ToString()
+    {
+        return Regex.Unescape(JsonSerializer.Serialize(this, JsonSerializeOptions.DefaultOptions));
+    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Speak/v1/WebSocket/v1/CommandSource.cs:22]: This value can change. Ignoring this comment.

File: Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs [hunk 8]
```diff
@@ -0,0 +1,875 @@
...
+        Log.Verbose("SpeakClient", "ENTER");
+
+        options ??= new DeepgramWsClientOptions(apiKey);
+        _deepgramClientOptions = options;
+
+        Log.Debug("SpeakClient", $"APIVersion: {options.APIVersion}");
+        Log.Debug("SpeakClient", $"BaseAddress: {options.BaseAddress}");
+        Log.Debug("SpeakClient", $"options: {options.OnPrem}");
+        Log.Verbose("SpeakClient", "LEAVE");
+    }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Speak/v1/WebSocket/v1/Client.cs:40]: Will do this in a future version.

File: Deepgram/Models/Speak/v1/AsyncResponse.cs [hunk 9]
```diff
@@ -4,22 +4,17 @@
...
+// This class provides the AsyncResponse implementation for the Deepgram API
+//
+// Deprecated: This class is deprecated. Use the namespace `Deepgram.Models.Speak.v1.REST` package instead.
+// This will be removed in a future release.
+//
+// This class frozen and no new functionality will be added.
+// *********** WARNING ***********
+/// </summary>
+[Obsolete("Please use Deepgram.Models.Speak.v1.REST.AsyncResponse instead", false)]
+public record AsyncResponse : REST.AsyncResponse
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Speak/v1/AsyncResponse.cs:18]: This is intentional. This is to allow users to still user those classes without needing to transition right away. Ignoring.

File: Deepgram/Models/Speak/v1/TextSource.cs [hunk 10]
```diff
@@ -4,21 +4,18 @@
...
+// This class provides the TextSource implementation for the Deepgram API
+//
+// Deprecated: This class is deprecated. Use the namespace `Deepgram.Models.Speak.v1.REST` package instead.
+// This will be removed in a future release.
+//
+// This class is frozen and no new functionality will be added.
+// *********** WARNING ***********
+/// </summary>
+[Obsolete("Please use Deepgram.Models.Speak.v1.REST.TextSource instead", false)]
+public class TextSource(string text) : REST.TextSource(text)
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Speak/v1/TextSource.cs:18]: This is intentional. This is to allow users to still user those classes without needing to transition right away. Ignoring.

**Review by SandraRodgers — APPROVED** [review]

File: examples/speak/websocket/simple/Program.cs [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+            // Initialize Library with default logging
+            // Normal logging is "Info" level
+            //Library.Initialize();
+            // OR very chatty logging
+            Library.Initialize(LogLevel.Debug); // LogLevel.Default, LogLevel.Debug, LogLevel.Verbose
+
+            Console.WriteLine("\n\nPress any key to stop and exit...\n\n\n");
+
+            // use the client factory with a API Key set with the "DEEPGRAM_API_KEY" environment variable
+            var clientOptions = new DeepgramWsClientOptions("", "api.beta.deepgram.com");
```
Comment (SandraRodgers) [inline examples/speak/websocket/simple/Program.cs:25]: beta url might need to be removed

Comment (dvonthenen) [reply examples/speak/websocket/simple/Program.cs:25]: will remove when I update all the examples

**Pull request merged**